compiler/rustc_infer/src/infer/opaque_types.rs

```diff
@@ -602,17 +609,17 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
 /// Returns `true` if `opaque_hir_id` is a sibling or a child of a sibling of `def_id`.
 ///
 /// Example:
-/// ```rust
+/// ```ignore HELP (why does this not work?)
```

Comment:
this might just be a bug??
```
error[E0308]: mismatched types
  --> src/infer/opaque_types.rs:623:16
   |
7  |         pub type Baz = impl Bar;
   |                        -------- the expected opaque type
...
12 |     fn f2() -> bar::Baz { /* ... */ }
   |        --      ^^^^^^^^ expected opaque type, found `()`
   |        |
   |        implicitly returns `()` as its body has no tail or `return` expression
   |
   = note: expected opaque type `Baz`
                found unit type `()`
```